giotto/homology/grids.py

```diff
@@ -0,0 +1,182 @@
+from ._utils import _pad_diagram
+from ..utils.validation import validate_params
+
+from ..externals.python import CubicalComplex
+
+
+class CubicalPersistence(BaseEstimator, TransformerMixin):
+    """`Persistence diagrams <https://giotto.ai/theory>`_ resulting from
+    `filtrations of Cubical complex <https://giotto.ai/theory>`_.
+
+    Given a `grayscale image <https://giotto.ai/theory>`_, information
+    about the appearance and disappearance of topological features
+    (technically, `homology classes <https://giotto.ai/theory>`_) of various
+    dimensions and at different scales is summarised in the corresponding
+    persistence diagram.
+
+    Parameters
+    ----------
+    homology_dimensions : iterable, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    coeff : int prime, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    infinity_values : float or None, default : ``None``
+        Which death value to assign to features which are still alive at
+        filtration value `np.inf`. ``None`` assigns the maximum pixel
+        values within all images passed to meth:`fit`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    See also
+    --------
+    VietorisRipsPersistence
+
+    Notes
+    -----
+    `Gudhi <https://github.com/GUDHI/gudhi-devel>`_ is used as a C++ backend
+    for computing Cubical persistent homology. Python bindings were modified
+    for performance.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] P. Dlotko, "Cubical complex", 2015; `GUDHI User and Reference Manual \
+    <http://gudhi.gforge.inria.fr/doc/latest/group__cubical__complex.html>`_.
+
+    """
+    _hyperparameters = {'infinity_values_': [numbers.Number],
+                        '_homology_dimensions': [list, [int, (0, np.inf)]],
+                        'coeff': [int, (2, np.inf)]}
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2,
+                 infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _gudhi_diagram(self, X):
+        cubical_complex = CubicalComplex(
+            dimensions=X.shape,
+            top_dimensional_cells=X.flatten(order="F"))
+        Xds = cubical_complex.persistence(homology_coeff_field=self.coeff,
```

Comment:
I personally prefer variable names to reveal intent, so ideally `Xds` would be something like `X_diagrams`. As a compromise, would `Xdgms` be acceptable?

Reply:
The compromise seems acceptable :)